ui/uistatemanager.cpp

```diff
@@ -167,10 +173,32 @@ void UIStateManager::restoreState()
     restoreWindowState();
     restoreSplitterState();
     restoreHeaderState();
+
+    // Allow custom state per end point
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+    if (m_customRestoreMethod.isValid())
+#endif
+    {
+        Q_ASSERT(!Endpoint::instance()->label().isEmpty());
+        m_stateSettings->beginGroup(Endpoint::instance()->label());
+        m_customRestoreMethod.invoke(m_widget, Q_ARG(QSettings *, m_stateSettings));
+        m_stateSettings->endGroup();
+    }
 }
 
 void UIStateManager::saveState()
 {
+    // Allow custom state per end point
+#if QT_VERSION >= QT_VERSION_CHECK(5, 0, 0)
+    if (m_customSaveMethod.isValid())
+#endif
+    {
+        Q_ASSERT(!Endpoint::instance()->label().isEmpty());
+        m_stateSettings->beginGroup(Endpoint::instance()->label());
```

Comment:
label() looks problematic as that can contain the PID IIUC, ie. we would constantly change the settings key here.